pallets/torus0/src/agent.rs

```diff
@@ -100,6 +100,12 @@ pub fn register<T: crate::Config>(
     )
     .map_err(|_| crate::Error::<T>::NotEnoughBalanceToRegisterAgent)?;
 
+    // Registration cost is sent to treasury
+    let _ = <T as crate::Config>::Currency::deposit_creating(
+        &<T as crate::Config>::Governance::dao_treasury_address(),
+        burn,
```

Comment:
I think it's possible to use the imbalance discarded on line 95 to deposit on the treasury instead of using `deposit_creating`, no?